externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -262,6 +262,11 @@ object SparkSQLEngine extends Logging {
     val rootDir = _sparkConf.getOption("spark.repl.classdir").getOrElse(getLocalDir(_sparkConf))
     val outputDir = Utils.createTempDir(prefix = "repl", root = rootDir)
     _sparkConf.setIfMissing("spark.sql.legacy.castComplexTypesToString.enabled", "true")
+    // SPARK-47911: we must set a value instead of leaving it as None, otherwise, we will get a
```

Comment:
Just asking a stupid question, why the parameter `spark.sql.binaryOutputStyle` is not designed to default to UTF-8 to avoid updating conf when using it.

Reply:
I guess it is because 

> the binary output format is inconsistent across different clients.

cc @yaooqinn

Reply:
Yes. Defined None as default, then different cli can further define their default behavior when meeting None